publication-commons/src/main/java/no/unit/nva/publication/UserAccessValidationUtil.java

```diff
@@ -0,0 +1,114 @@
+import no.unit.nva.model.Identity;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.PublicationStatus;
+import no.unit.nva.model.Reference;
+import no.unit.nva.model.instancetypes.PublicationInstance;
+import no.unit.nva.model.instancetypes.degree.DegreeBachelor;
+import no.unit.nva.model.instancetypes.degree.DegreeMaster;
+import no.unit.nva.model.instancetypes.degree.DegreePhd;
+import no.unit.nva.model.pages.Pages;
+import no.unit.nva.publication.model.business.UserInstance;
+import nva.commons.apigateway.AccessRight;
+import nva.commons.apigateway.RequestInfo;
+import nva.commons.apigateway.exceptions.UnauthorizedException;
+
+public class UserAccessValidationUtil {
+
+    public static void validateUserAccessForDeletingPublishedPublication(RequestInfo requestInfo,
+                                                                        Publication publication)
+        throws UnauthorizedException {
+
+        if (!PublicationStatus.PUBLISHED.equals(publication.getStatus())) {
+            return;
+        }
+
+        if (isAuthorizedToUnpublishPublication(requestInfo, publication)) {
+            return;
+        }
+
+        throw new UnauthorizedException();
+    }
+
+    private static boolean isAuthorizedToUnpublishPublication(RequestInfo requestInfo, Publication publication)
+        throws UnauthorizedException {
+
+        return isEditor(publication, requestInfo)
+               || canEditResourcesForInstitution(publication, requestInfo)
+               || isPublicationOwner(publication, requestInfo.getUserName())
+               || isContributor(publication, requestInfo.getPersonCristinId());
+    }
+
+    private static boolean isEditor(Publication publication, RequestInfo requestInfo) {
+        if (isDegree(publication)) {
+            return hasAccessRight(requestInfo, PUBLISH_DEGREE);
+        }
+
+        return hasAccessRight(requestInfo, EDIT_ALL_NON_DEGREE_RESOURCES);
+    }
+
+    private static boolean isDegree(Publication publication) {
+        return Optional.ofNullable(publication.getEntityDescription())
+                   .map(EntityDescription::getReference)
+                   .map(Reference::getPublicationInstance)
+                   .map(UserAccessValidationUtil::publicationInstanceIsDegree)
+                   .orElse(false);
+    }
+
+    private static Boolean publicationInstanceIsDegree(PublicationInstance<? extends Pages> publicationInstance) {
+        return publicationInstance instanceof DegreeBachelor
+               || publicationInstance instanceof DegreeMaster
+               || publicationInstance instanceof DegreePhd;
+    }
+
+    private static boolean canEditResourcesForInstitution(Publication publication, RequestInfo requestInfo)
+        throws UnauthorizedException {
+
+        if (!userIsFromSameInstitutionAsPublication(requestInfo, publication)) {
+            return false;
+        }
+
+        return hasAccessRight(requestInfo, EDIT_OWN_INSTITUTION_RESOURCES);
+    }
```

Comment:
Dette er en metode som inneholder logikk som vi bruker flere steder. Kanskje på tide til å implementere en metode i requestInfo som tar inn publikasjon som input, for eksempel:

requestInfo.isAuthorizedAtCustomer(accessRight, entity)
 eller 
requestInfo.hasSameCustomer(entity).

Hvor entity er Publication/Ticket/kanskje message? @brinxmat